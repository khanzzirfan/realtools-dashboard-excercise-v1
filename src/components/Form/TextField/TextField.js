import { Field, Label, Hint } from '../FieldParts';
import { Input } from './TextField.styled';
import PropTypes from 'prop-types';

export default function TextField({ id, label, helper, errorMsg, ...props }) {
  const invalidAttr = errorMsg ? { 'aria-invalid': true } : {};

  return (
    <Field as="label">
      <Label htmlFor={id}>{label}</Label>
      <Input id={id} {...props} {...invalidAttr} />
      {errorMsg && <Hint errorMsg={errorMsg} helper={helper} />}
    </Field>
  );
}

TextField.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  helper: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  errorMessage: PropTypes.string,
};

TextField.defaultProps = {
  type: 'text',
};
