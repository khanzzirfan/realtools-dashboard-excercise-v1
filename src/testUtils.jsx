import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { AppContextProvider } from './hooks/useAppContext';
import { theme } from './theme';

const AllTheProviders = ({ children }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        // âœ… turns retries off
        retry: 2,
      },
    },
  });
  return (
    <QueryClientProvider client={queryClient}>
      <AppContextProvider>
        <ThemeProvider theme={theme}>
          <MemoryRouter initialEntries={['/people', '/people/new', '/people/edit/:id']}>
            {children}
          </MemoryRouter>
        </ThemeProvider>
      </AppContextProvider>
    </QueryClientProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
