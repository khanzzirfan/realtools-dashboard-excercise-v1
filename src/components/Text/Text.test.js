import '@testing-library/jest-dom';
import { render } from '../../testUtils';
import TextField, { TextLight } from './Text';

describe('TextField', () => {
  it('should renders correctly as span element and h1 typography styles', () => {
    const { asFragment } = render(
      <TextField as="span" size="h1">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as span element and h2typography styles', () => {
    const { asFragment } = render(
      <TextField as="span" size="h2">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as span element and h3 typography styles', () => {
    const { asFragment } = render(
      <TextField as="span" size="h3">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as span element and h4 typography styles', () => {
    const { asFragment } = render(
      <TextField as="span" size="h4">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodyLead typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodyLead">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and body typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="body">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodyMedium typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodyMedium">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodyBold typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodyBold">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodySmall typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodySmall">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodySmallBold typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodySmallBold">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodySmallMedium typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodySmallMedium">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders correctly as p element and bodyCaption typography styles', () => {
    const { asFragment } = render(
      <TextField as="p" size="bodyCaption">
        Contractor
      </TextField>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders light text as p element', () => {
    const { asFragment } = render(
      <TextLight as="p" size="bodyCaption">
        Contractor
      </TextLight>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
