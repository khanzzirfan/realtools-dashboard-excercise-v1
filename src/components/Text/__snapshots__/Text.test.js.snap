// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should renders correctly as p element and body typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodyBold typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.5;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodyCaption typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 400;
  font-size: 0.8125rem;
  line-height: 1.23;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodyLead typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 400;
  font-size: 1.125rem;
  line-height: 1.44;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodyMedium typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.5;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodySmall typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.29;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodySmallBold typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 600;
  font-size: 0.875rem;
  line-height: 1.29;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as p element and bodySmallMedium typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.58;
  margin: 0;
}

<p
    class="c0"
  >
    Contractor
  </p>
</DocumentFragment>
`;

exports[`TextField should renders correctly as span element and h1 typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 500;
  font-size: 1.875rem;
  line-height: 1.34;
  margin: 0;
}

<span
    class="c0"
  >
    Contractor
  </span>
</DocumentFragment>
`;

exports[`TextField should renders correctly as span element and h2typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 500;
  font-size: 1.625rem;
  line-height: 1.23;
  margin: 0;
}

<span
    class="c0"
  >
    Contractor
  </span>
</DocumentFragment>
`;

exports[`TextField should renders correctly as span element and h3 typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 500;
  font-size: 1.5rem;
  line-height: 1.25;
  margin: 0;
}

<span
    class="c0"
  >
    Contractor
  </span>
</DocumentFragment>
`;

exports[`TextField should renders correctly as span element and h4 typography styles 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 500;
  font-size: 1.375rem;
  line-height: 1.28;
  margin: 0;
}

<span
    class="c0"
  >
    Contractor
  </span>
</DocumentFragment>
`;

exports[`TextField should renders light text as p element 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 400;
  font-size: 0.8125rem;
  line-height: 1.23;
  margin: 0;
}

.c1 {
  color: var(--colors-lynch);
}

<p
    class="c0 c1"
  >
    Contractor
  </p>
</DocumentFragment>
`;
